extensions/panache/mongodb-panache/deployment/src/main/java/io/quarkus/mongodb/panache/deployment/PanacheResourceProcessor.java

```diff
@@ -214,7 +227,17 @@ void buildMutiny(CombinedIndexBuildItem index,
             ApplicationIndexBuildItem applicationIndex,
             BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
             BuildProducer<PropertyMappingClassBuildStep> propertyMappingClass,
-            BuildProducer<BytecodeTransformerBuildItem> transformers) {
+            BuildProducer<BytecodeTransformerBuildItem> transformers) throws BuildException {
+
+        // we verify that no ID fields are defined (via @BsonId) when extending ReactivePanacheMongoEntity
+        for (AnnotationInstance annotationInstance : index.getIndex().getAnnotations(DOTNAME_BSON_ID)) {
+            ClassInfo info = io.quarkus.panache.common.deployment.JandexUtil.getEnclosingClass(annotationInstance);
+            if (info.superName().equals(DOTNAME_MUTINY_PANACHE_ENTITY)) {
```

Comment:
Same remark about inheritance.

Reply:
Done, I climb up the hierarchy of classes.